src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -92,20 +89,37 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
             if (Url.IsLocalUrl(returnUrl))
                 return Redirect(returnUrl);
 
-            if (string.IsNullOrEmpty(returnUrl))
+            if (string.IsNullOrWhiteSpace(returnUrl))
                 return Redirect("~/");
 
-            // User might have clicked on a malicious link - should be logged
-            throw new Exception("Invalid return URL");
+            // Return URL could be malicious
+            throw new InvalidReturnUrlException();
         }
 
-        [HttpGet]
         public async Task<IActionResult> Error(string errorId)
         {
             // retrieve error details from identityserver
             var message = await _interaction.GetErrorContextAsync(errorId);
 
-            return Ok(message);
+            return View(new ErrorViewModel { Message = message?.ErrorDescription });
```

Comment:
As `errorId` will only ever be `null` at this stage of development, the same applies to `message`. Would we ever realistically want to display this error information?

Reply:
We'd probably want to provide the error message in development builds? Maybe?

Reply:
We might not want to show it but we would definitely want to log it. Sometimes you might want to show an error message if it helps the user. Not sure we have a case for it yet.

Reply:
Primarily thinking from a security point-of-view – I've not actually seen a real example message yet but if it reveals details of the security implementation then we obviously wouldn't want to display it.

Reply:
As we don't have a case for it right now, I'll remove it. We can reinstate it later if required.